lib/Doctrine/ORM/Tools/Pagination/LimitSubqueryOutputWalker.php

```diff
@@ -85,13 +110,27 @@ public function __construct($query, $parserResult, array $queryComponents)
      * Walks down a SelectStatement AST node, wrapping it in a SELECT DISTINCT.
      *
      * @param SelectStatement $AST
+     * @param bool $addMissingItemsFromOrderByToSelect
      *
      * @return string
      *
      * @throws \RuntimeException
      */
-    public function walkSelectStatement(SelectStatement $AST)
+    public function walkSelectStatement(SelectStatement $AST, $addMissingItemsFromOrderByToSelect = true)
```

Comment:
The reason for this is because the walker clones itself and then calls walkSelectStatement on the clone. The method that does the cloning and that nonsense is called from walkSelectStatement. Not sure how better to structure that call chain so as to avoid recursion and still allow the functionality.